File: api-asic/csharp/src/core/SignableASiCDocument.cs
/**
 * @author yavuz.kahveci
 */
using System.IO;
using tr.gov.tubitak.uekae.esya.api.asic.model;
using tr.gov.tubitak.uekae.esya.api.signature.impl;

namespace tr.gov.tubitak.uekae.esya.api.asic.core
{
    public class SignableASiCDocument : BaseSignable
    {
        readonly ASiCDocument document;

        public SignableASiCDocument(ASiCDocument aDocument)
        {
            document = aDocument;
        }

        public override Stream getContent()
        {
            MemoryStream baos = new MemoryStream();
            document.write(baos);
            content = new MemoryStream(baos.ToArray());
            return content;
        }

        public override string getURI()
        {
            return document.getASiCDocumentName();
        }

        public override string getMimeType()
        {
            return "text/xml";
        }
    }
}

File: api-asic/src/tr/gov/tubitak/uekae/esya/api/asic/core/SignableASiCDocument.java
package tr.gov.tubitak.uekae.esya.api.asic.core;

import tr.gov.tubitak.uekae.esya.api.asic.model.ASiCDocument;
import tr.gov.tubitak.uekae.esya.api.common.util.StreamUtil;
import tr.gov.tubitak.uekae.esya.api.signature.SignatureException;
import tr.gov.tubitak.uekae.esya.api.signature.impl.BaseSignable;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.InputStream;

/**
 * @author ayetgin
 */
public class SignableASiCDocument extends BaseSignable
{
    ASiCDocument document;

    public SignableASiCDocument(ASiCDocument aDocument)
    {
        document = aDocument;
    }

    public InputStream getContent() throws SignatureException
    {
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        document.write(baos);
        return new ByteArrayInputStream(baos.toByteArray());
    }

    public String getURI()
    {
        return document.getASiCDocumentName();
    }

    public String getMimeType()
    {
        return "text/xml";
    }
}

